### PR TITLE
Bug #18096: Relationships can no longer be committed without a roleplayer

### PR DESCRIPTION
**Pull request by fppt**
This adds a new validation rule and cache which makes sure we can't commit relationships without a role player

### CODE REVIEW

File: grakn-kb/src/test/java/ai/grakn/kb/internal/concept/RelationshipTest.java [hunk 1]
```diff
@@ -311,4 +314,17 @@ public void whenAttributeLinkedToRelationshipIsInferred_EnsureItIsMarkedAsInferr
         RelationshipImpl.from(relationship).attributeInferred(attribute);
         assertTrue(relationship.relationships().findAny().get().isInferred());
     }
+
+    @Test
+    public void whenAddingRelationshipWithNoRolePlayers_Throw(){
```
Comment (lolski) [inline grakn-kb/src/test/java/ai/grakn/kb/internal/concept/RelationshipTest.java]: Just double checking, do we already have a test which verifies the happy path?

File: grakn-kb/src/main/java/ai/grakn/kb/internal/ValidateGlobalRules.java [hunk 2]
```diff
@@ -388,4 +388,15 @@ private static ReasonerQuery combinedRuleQuery(GraknTx graph, Rule rule){
 
         return errors;
     }
+
+    /**
+     * Checks if a {@link Relationship} has at least one role player.
+     * @param relationship The {@link Relationship} to check
+     */
+    static Optional<String> validateRelationshipHasRolePlayers(Relationship relationship) {
```
Comment (lolski) [inline grakn-kb/src/main/java/ai/grakn/kb/internal/ValidateGlobalRules.java]: Out of curiosity, is it ok to have a relationship with, let's say, 3 roles of which only 1 has a role player?

Comment (kasper-piskorski) [reply grakn-kb/src/main/java/ai/grakn/kb/internal/ValidateGlobalRules.java]: Yes. You can have a single role-roleplayer mapping for a given relation instance plus you can also have a single roleplayer playing multiple roles.

File: grakn-kb/src/test/java/ai/grakn/kb/internal/concept/RelationshipTest.java (hunk 1 above)
Comment (fppt) [reply grakn-kb/src/test/java/ai/grakn/kb/internal/concept/RelationshipTest.java]: Yep we do.

**Pull request merged**